classes/local/report.php

```diff
@@ -5389,9 +5389,9 @@ public function get_content($tab = null, $ptutor = null, $padmin = null, $arch =
                                     }
                                 }
 
-                                // If there are any comments or other feedback (such as online PDF
+                                // If there is a grade and any comments or other feedback (such as online PDF
                                 // files, rubrics or marking guides).
-                                if ($record->feedbacklink || $onlinepdffeedback || $feedbacktext) {
+                                if ($record->grade && ($record->feedbacklink || $onlinepdffeedback || $feedbacktext)) {
```

Comment:
does mod_assign not have proper fucntion you can call to find out if something is meant to be released to students?

Having a grade in the gradebook is a good proxy but seems odd there isn't a better way

Reply:
How is that?

Reply:
isn't there like a "Feedback released to students" type status / "is feedback released? yes/no" function we could call

Reply:
@opitz was this extra permission check needed ?

doesn't really change the timing of the display of the feedback in any way

Reply:
@aspark21 I am not exactly sure what you are referring to.
I added extra checks about capabilities because the myfeedback report may as well be used by a tutor or admin, in which case the current user is not the student and may have no grade which in turn will fail to present any feedback for the student to the tutor or the admin - an issue that was revealed by a failing Behat test.
The extra check make sure that a tutor or admin will be able to see any feedbacks while the student will not see them.

Reply:
maybe go back to original @opitz